MozillaFoundation/foundation.mozilla.org
# DonateHelpPage model, factory, and tests

Related Ticket: #11172

Link to review page: https://donate-foundation-s-11177-dona-5stlkm.mofostaging.net/en/help/


# Description

Originally, there were only two page models related to donations on the foundation stack. These were:
- `BaseDonationPage` - Which serves as the base page model, in case we need to add any shared fields/functionality across Donation related models. (currently, there are none so its pretty much blank.)
- `DonationLandingPage` - Which serves as the landing page for the donation related pages.

These both lived in a file `models.py`, which goes against what we do for other models in our codebase. 

If you take a look at the folder `wagtailpages` for example, you will see that `models.py` imports all page models, which all live under their respective folders/files. (for example `blog/blog_page.py`)

Since we are so early into the development of the donate side of the foundation site, I thought that this PR would be the best time to do some refactoring so the donate models can fall under the same convention. 

That is the reason that alongside the `DonateHelpPage` model changes, you will also see the following additional file changes:

- `pagemodels/__init__.py`
- `pagemodels/base_page.py`
- `pagemodels/landing_page.py`

The code related to these page models has already been approved in [this PR](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/10401), so I do not believe there is a need to look too into these.


**That being said, this PR introduces the following DonateHelpPage related updates:**

- A new model `DonateHelpPage` which at the moment only has one  `StreamField` field `body`. The streamblocks available for use are the same as `CampaignPage` per request of design. (They both use our pre-defined list called `base_fields`)
- A template for the new model, which at the moment only renders the page's title, as well as placeholders for "Intro", "Form", and "Body"
- A new factory for the model, and an update to existing factories so new db's will include a help page titled "Donate Help"
- A basic test suite for the new model, as well as one small update to the `DonateLandingPage` tests, confirming that it can create a `DonateHelpPage`.



# Screenshots
![Screenshot 2023-09-27 at 15-40-58 Donate Help](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/18314510/18993d89-3c16-419f-96d0-3ff074d9e72b)



# Steps to test

1. Checkout this branch
2. Run `inv new-db`
3. Log into the CMS
4. Alongside "Homepage" and "Mozilla Festival", there should be a page titled "Donate Now" with a subpage "Donate Help".
5. Click into "Donate Help" and take a look around, it should only have two fields "title" and "body".
6. Visit the page on the front end, it should look like the screenshot above.
7. If everything is working as expected, testing is complete!
